service/src/test/java/bio/terra/axonserver/app/controller/CromwellWorkflowControllerTest.java

```diff
@@ -0,0 +1,220 @@
+package bio.terra.axonserver.app.controller;
+
+import static bio.terra.axonserver.testutils.MockMvcUtils.USER_REQUEST;
+
+import bio.terra.axonserver.model.ApiWorkflowIdAndLabel;
+import bio.terra.axonserver.model.ApiWorkflowIdAndStatus;
+import bio.terra.axonserver.model.ApiWorkflowMetadataResponse;
+import bio.terra.axonserver.model.ApiWorkflowQueryResponse;
+import bio.terra.axonserver.model.ApiWorkflowQueryResult;
+import bio.terra.axonserver.service.cromwellworkflow.CromwellWorkflowService;
+import bio.terra.axonserver.service.wsm.WorkspaceManagerService;
+import bio.terra.axonserver.testutils.BaseUnitTest;
+import bio.terra.axonserver.testutils.MockMvcUtils;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.cromwell.client.ApiException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.client.model.CromwellApiLabelsResponse;
+import io.swagger.client.model.CromwellApiWorkflowIdAndStatus;
+import io.swagger.client.model.CromwellApiWorkflowMetadataResponse;
+import io.swagger.client.model.CromwellApiWorkflowQueryResponse;
+import io.swagger.client.model.CromwellApiWorkflowQueryResult;
+import java.util.Date;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+public class CromwellWorkflowControllerTest extends BaseUnitTest {
+  @Autowired private MockMvcUtils mockMvcUtils;
+  @Autowired private MockMvc mockMvc;
+  @Autowired private ObjectMapper objectMapper;
+
+  @MockBean private CromwellWorkflowService cromwellWorkflowService;
+  @MockBean private WorkspaceManagerService wsmService;
+
+  private final UUID workspaceId = UUID.randomUUID();
+  private final UUID workflowId = UUID.randomUUID();
+  private final String DEFAULT_WORKFLOW_STATUS = "Submitted";
+  private final Date DEFAULT_WORKFLOW_SUBMISSION_DATE = new Date(0);
+
+  private final String CROMWELL_WORKFLOW_STATUS_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/status";
+  private final String CROMWELL_WORKFLOW_LABELS_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/labels";
+  private final String CROMWELL_WORKFLOW_METADATA_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/%s/metadata";
+  private final String CROMWELL_WORKFLOW_QUERY_PATH_FORMAT =
+      "/api/workspaces/%s/cromwell/workflows/query";
+
+  @Test
+  void workspaceAccess_or_labelValidation_failure() throws Exception {
+    Mockito.doThrow(new ApiException(403, "No workspace access"))
+        .when(cromwellWorkflowService)
+        .validateWorkspaceAccessAndWorkflowLabelMatches(
+            workflowId, workspaceId, USER_REQUEST.getToken());
+
+    // Stub the client status response.
+    Mockito.when(cromwellWorkflowService.getStatus(workflowId))
```

Comment:
No, we don't call cromwell. 

I mocked it since we were throwing 500 before, so this test would always throw 500 (since the response didn't serialize correctly). We can throw a 403 code instead, and remove this mock call.